Compile bug: Build failure on ppc64le from simd-mappings.h

### Git commit

995083e


### Operating systems

Linux

### GGML backends

CPU

### Problem description & steps to reproduce

While building llama.cpp on Power (ppc64le) with gcc13, seeing below errors:

llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]

### First Bad Commit

_No response_

### Compile command

```shell
cmake –-build build_llama –-config Release
make
```

### Relevant log output

```shell
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
llama.cpp/ggml/src/ggml-cpu/simd-mappings.h:395:58: error: lvalue required as unary ‘&’ operand
  395 | #define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
```

--- a/common/arg.cpp
+++ b/common/arg.cpp
@@ -163,6 +163,8 @@ struct common_hf_file_res {
 #   if !defined(PATH_MAX)
 #   define PATH_MAX MAX_PATH
 #   endif
+#elif defined(_AIX)
+#include <sys/limits.h>
 #else
 #include <sys/syslimits.h>
 #endif

--- a/ggml/src/ggml-cpu/simd-mappings.h
+++ b/ggml/src/ggml-cpu/simd-mappings.h
@@ -392,7 +392,11 @@ static inline void __avx_f32cx8_store(ggml_fp16_t *x, __m256 y) {
 #define GGML_F16_VEC_LOAD(p, i) (i & 0x1) ?                   \
   vec_extract_fp32_from_shorth(vec_xl(0, p - GGML_F16_EPR)) : \
   vec_extract_fp32_from_shortl(vec_xl(0, p))
-#define GGML_ENDIAN_BYTE(i) ((unsigned char *)&(uint16_t){1})[i]
+static inline unsigned char ggml_endian_byte(int i) {
+       uint16_t tmp_val = 1;
+       return ((unsigned char *)&tmp_val)[i];
+}
+#define GGML_ENDIAN_BYTE(i) ggml_endian_byte(i)
 #define GGML_F16_VEC_STORE(p, r, i)                             \
   if (i & 0x1)                                                  \
     vec_xst(vec_pack_to_short_fp32(r[i - GGML_ENDIAN_BYTE(1)],  \

